File: intan_pariwara/intan_pariwara/custom/lead.js
// Copyright (c) 2025, DAS and contributors
// For license information, please see license.txt

frappe.ui.form.on("Lead", {
    annual_revenue_dana_bos: function (frm) {
        frm.trigger("annual_revenue_potensi")
    },
    _annual_revenue_dana_bos: function (frm) {
        frm.trigger("annual_revenue_potensi")
    },
    annual_revenue_potensi_dana_siswa: function (frm) {
        frm.trigger("annual_revenue_potensi")
    },
    annual_revenue_potensi: function (frm) {
        var potensi_bos = flt(frm.doc.annual_revenue_dana_bos * frm.doc._annual_revenue_dana_bos / 100)

        frm.set_value("annual_revenue_potensi_bos", potensi_bos)
        frm.set_value("total_annual_revenue_potensi",flt(frm.doc.annual_revenue_potensi_dana_siswa + potensi_bos))
    } 
})

File: intan_pariwara/intan_pariwara/custom/lead.py
# Copyright (c) 2025, DAS and Contributors
# License: GNU General Public License v3. See license.txt

import frappe
from frappe.utils import flt

def set_annual_revenue_potensi(self, method):
    self.annual_revenue_potensi_bos = flt(self.annual_revenue_dana_bos * self._annual_revenue_dana_bos / 100)
    self.total_annual_revenue_potensi = flt(self.annual_revenue_potensi_dana_siswa + self.annual_revenue_potensi_bos)